dynamic_analysis_pkg/dynamic_analysis_pkg/__init__.py

```diff
@@ -0,0 +1,167 @@
+# pylint: disable=missing-function-docstring
+'''Machinary to fetch simulation results to a Jupyter Notebook'''
+from pathlib import Path, PurePath
+from shutil import move
+from datetime import datetime
+import logging
+import os
+import pandas as pd
+from tabulate import tabulate
+import pyunicore.client as unicore_client
+from tqdm.notebook import tqdm
+
+L = logging.getLogger('dynamic-analysis-pkg')
+L.setLevel(logging.INFO)
+
+UNICORE_ENDPOINT = 'https://bspsa.cineca.it/advanced/pizdaint/rest/core'
+
+
+class AResults:
+    '''
+    Based on a simulation id, fetch the results using Unicore API
+    '''
+    DEFAULT_CIRCUIT_DIR = '/home/data-bbp/20191017/'
+    DEFAULT_WD_BASE = '/home/simulation-results'
+    ORIGIN_CIRCUIT_PATH = '/store/hbp/ich002/antonel/O1/20191017/'
+    HPC_DIR_BASE = '/scratch/snx3000/unicore/FILESPACE/'
+    JOB_BASE = os.path.join(UNICORE_ENDPOINT, 'jobs')
+
+    def __init__(self, token, sim_id, base_working_directory=None, new_circuit_path=None):
+        self.hpc_dir_path = None
+        self.files_list = None
+        self.circuit_dir = None
+        self.local_dir = None
+
+        if base_working_directory is None:
+            base_working_directory = self.DEFAULT_WD_BASE
+        if new_circuit_path is None:
+            new_circuit_path = self.DEFAULT_CIRCUIT_DIR
+
+        self.define_paths(sim_id, base_working_directory, new_circuit_path)
+        self.fetch_results(token, sim_id)
+        self.blueconfig = Path(self.local_dir) / 'BlueConfig'
+
+    def fetch_results(self, token, sim_id=None):
+        if not sim_id:
+            raise ValueError('[Error] Simulation ID not provided')
+
+        self.retrieve_sim_info(token, sim_id)
+        self.get_sim_results()
+
+    def retrieve_sim_info(self, token, sim_id=None):
+        tr = unicore_client.Transport(token)
+        job_url = os.path.join(self.JOB_BASE, sim_id)
```

Comment:
Since in your next patchset, `self.JOB_BASE` will be a `Path`:
```suggestion
        job_url = self.JOB_BASE / sim_id
```

Reply:
For this one I have a question..
If I use Path with and URL (in this case UNICORE_ENDPOINT) it removes `//` and set `/` and that will mess up the API call, that is why I was using the `os.path.join` do you know any other way?

Reply:
I see. Yes, I think you are right. I don't think `Path` is intended to be used with URLs.